hare/interfaces.go

```diff
@@ -27,6 +27,14 @@ type stateQuerier interface {
 	IsIdentityActiveOnConsensusView(context.Context, types.NodeID, types.LayerID) (bool, error)
 }
 
-type nonceFetcher interface {
+type mesh interface {
```

Comment:
testing with a real database is a PITA.

i tolerated it before because before this PR it only needs to generate/save proposals into database.

but now that i have to create the FULL atx for testing, while all i need is basetickcount and tickcount in the header.
@fasmat will sympathize i am sure.

besides, with the real database you don't get to test errors other than `not found` error. tho that's a very small point.

Reply:
So is the reason that you need to generate a full atx for testing because yo need to control the tick heights?

Why not just mock the CacheDB? This mesh construction seems a bit odd to me, its like a little addition to the db, it calls itself mesh, but it seems like its just a db

Reply:
what is the advantage of mocking CacheDB over this?
- i still need to create a (minimal) interface on the hare (client) side for testing.
- CacheDB is mostly used to cache ATXs/nonce/maliciousness of identities, which will be query a lot during hare. it doesn't have interface for proposals and ballots.

mesh is a general term for mesh data and a abstraction of access to data. it could be that we need to fetch data from peers. hare proper should not care.

Reply:
I wouldn't say there is an advantage to mocking CacheDB over the mesh. But I don't think (with my limited knowledge of the codebase) that the mesh stands alone as a particularly useful construct. If the way that you want to access the database is the current mesh interface, why not just make the database have that interface? I don't see any reason to split the tiny amount of logic that the mesh contains from the database.

Having the mesh now requires implementing several methods that provide no value because they are simply delegating to the database. E.g.
```
func (m defaultMesh) Header(id types.ATXID) (*types.ActivationTxHeader, error) {
	return m.cdb.GetAtxHeader(id)
}
```

Reply:
make sense to get rid of the delegation. done.

Reply:
It goes into ownership of the interface. IMO the interface should be designed with the needs of its user in mind, not with the API of its provider.

Just because the API of the database changes we don't want to have to update Hare, that's why it often makes sense to introduce a custom interface and a small adapter.

The adapter is cheap and if the database wasn't our code but a dependency we are using for me absolute the way to go. But because we have full control over Hare and the DB API it depends on our anticipation of what will be more a PITA in the future:

a) having to update hare in multiple places because we had to update the API of the DB and updating all mocks / tests because the implementation drives the interface or
b) only having to update the adapter because the user of the interface drives it design and the implementation has to be adapted to it?